### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by oswetto**

### CODE REVIEW

**Comment by gbrail** [comment]
Thanks for working on improving the codebase!

It looks like this PR mainly adds SerialVersionUIDs where they're missing. However, for backward compatibility and general goodness, I think we should follow the existing pattern in the codebase of including the generated UID (as generated by "serialver" or your IDE) rather than setting them all to 1.

Also it looks like the Checkstyle checks fail in the Jenkins build. You can check these yourself by running "./gradlew check".

File: src/org/mozilla/javascript/NativeError.java [hunk 1]
```diff
@@ -379,42 +379,12 @@ protected int findPrototypeId(String s)
         private int stackTraceLimit = DEFAULT_STACK_LIMIT;
         private Function prepareStackTrace;
 
-        public Object getStackTraceLimit(Scriptable thisObj) {
```
Comment (gbrail) [inline src/org/mozilla/javascript/NativeError.java]: Why remove this function (and the next three)? I'm pretty sure that they're part of the stack trace customization features, if I recall.

Comment (oswetto) [reply src/org/mozilla/javascript/NativeError.java]: I had seen that the methods are not used, I'll add at next commit

File: src/org/mozilla/javascript/NativeStringIterator.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 package org.mozilla.javascript;
 
 public final class NativeStringIterator extends ES6Iterator {
-    private static final long serialVersionUID = 1L;
+    private static final long serialVersionUID = 5890115206182483238L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/NativeStringIterator.java]: Technically this breaks backward compatibility. Rhino is used in a lot of places and we never know. Can you take this one out?

File: src/org/mozilla/javascript/ScriptRuntime.java [hunk 3]
```diff
@@ -2167,19 +2167,18 @@ public static Object setConst(Scriptable bound, Object value,
      * to see if a given property has already been enumerated.
      *
      */
-    private static class IdEnumeration implements Serializable
-    {
-        private static final long serialVersionUID = 1L;
+    private static class IdEnumeration implements Serializable {
+        private static final long serialVersionUID = 3052885577094294244L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/ScriptRuntime.java]: Same here -- the class already had a UID so if we change it then we break compatibility.

File: src/org/mozilla/javascript/ast/AstNode.java [hunk 4]
```diff
@@ -134,9 +132,8 @@
     }
 
     public static class PositionComparator implements Comparator<AstNode>, Serializable {
-        private static final long serialVersionUID = 1L;
-
-        /**
+        private static final long serialVersionUID = 1551233591343412911L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/ast/AstNode.java]: Another compatibility one.

File: src/org/mozilla/javascript/commonjs/module/ModuleScope.java [hunk 5]
```diff
@@ -15,9 +15,7 @@
  * and check sandbox constraints.
  */
 public class ModuleScope extends TopLevel {
-
-    private static final long serialVersionUID = 1L;
-
+    private static final long serialVersionUID = -15979856443615558L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/ModuleScope.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/ModuleScript.java [hunk 6]
```diff
@@ -16,10 +16,8 @@
  * @author Attila Szegedi
  * @version $Id: ModuleScript.java,v 1.3 2011/04/07 20:26:11 hannes%helma.at Exp $
  */
-public class ModuleScript implements Serializable
-{
-    private static final long serialVersionUID = 1L;
-
+public class ModuleScript implements Serializable {
+    private static final long serialVersionUID = -8287890021565388770L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/ModuleScript.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/Require.java [hunk 7]
```diff
@@ -38,10 +38,8 @@
  * @author Attila Szegedi
  * @version $Id: Require.java,v 1.4 2011/04/07 20:26:11 hannes%helma.at Exp $
  */
-public class Require extends BaseFunction
-{
-    private static final long serialVersionUID = 1L;
-
+public class Require extends BaseFunction {
+    private static final long serialVersionUID = 1880457728609216212L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/Require.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/RequireBuilder.java [hunk 8]
```diff
@@ -20,10 +20,8 @@
  * @author Attila Szegedi
  * @version $Id: RequireBuilder.java,v 1.4 2011/04/07 20:26:11 hannes%helma.at Exp $
  */
-public class RequireBuilder implements Serializable
-{
-    private static final long serialVersionUID = 1L;
-
+public class RequireBuilder implements Serializable {
+    private static final long serialVersionUID = -8686999731177566941L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/RequireBuilder.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/CachingModuleScriptProviderBase.java [hunk 9]
```diff
@@ -23,13 +23,9 @@
...
  */
-public abstract class CachingModuleScriptProviderBase
-implements ModuleScriptProvider, Serializable
-{
-    private static final long serialVersionUID = 1L;
-
-    private static final int loadConcurrencyLevel =
-        Runtime.getRuntime().availableProcessors() * 8;
+public abstract class CachingModuleScriptProviderBase implements ModuleScriptProvider, Serializable {
+    private static final long serialVersionUID = -3212148720202920385L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/CachingModuleScriptProviderBase.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/DefaultUrlConnectionExpiryCalculator.java [hunk 10]
```diff
@@ -15,11 +15,8 @@
  * @author Attila Szegedi
  * @version $Id: DefaultUrlConnectionExpiryCalculator.java,v 1.3 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public class DefaultUrlConnectionExpiryCalculator
-implements UrlConnectionExpiryCalculator, Serializable
-{
-    private static final long serialVersionUID = 1L;
-
+public class DefaultUrlConnectionExpiryCalculator implements UrlConnectionExpiryCalculator, Serializable {
+    private static final long serialVersionUID = 6649125723317726741L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/DefaultUrlConnectionExpiryCalculator.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/ModuleSource.java [hunk 11]
```diff
@@ -30,10 +30,8 @@
  * @author Attila Szegedi
  * @version $Id: ModuleSource.java,v 1.3 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public class ModuleSource implements Serializable
-{
-    private static final long serialVersionUID = 1L;
-
+public class ModuleSource implements Serializable {
+    private static final long serialVersionUID = 7183957145984428890L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/ModuleSource.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/ModuleSourceProviderBase.java [hunk 12]
```diff
@@ -26,10 +26,8 @@
  * @author Attila Szegedi
  * @version $Id: ModuleSourceProviderBase.java,v 1.3 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public abstract class ModuleSourceProviderBase implements
-        ModuleSourceProvider, Serializable
-{
-    private static final long serialVersionUID = 1L;
+public abstract class ModuleSourceProviderBase implements ModuleSourceProvider, Serializable {
+    private static final long serialVersionUID = -5687971989835149396L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/ModuleSourceProviderBase.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/ParsedContentType.java [hunk 13]
```diff
@@ -13,10 +13,8 @@
  * @author Attila Szegedi
  * @version $Id: ParsedContentType.java,v 1.3 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public final class ParsedContentType implements Serializable
-{
-    private static final long serialVersionUID = 1L;
-
+public final class ParsedContentType implements Serializable {
+    private static final long serialVersionUID = 2660291075921848672L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/ParsedContentType.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/SoftCachingModuleScriptProvider.java [hunk 14]
```diff
@@ -29,16 +29,10 @@
...
-    private static final long serialVersionUID = 1L;
-
-    private transient ReferenceQueue<Script> scriptRefQueue =
-        new ReferenceQueue<Script>();
-
-    private transient ConcurrentMap<String, ScriptReference> scripts =
-        new ConcurrentHashMap<String, ScriptReference>(16, .75f,
-                getConcurrencyLevel());
+public class SoftCachingModuleScriptProvider extends CachingModuleScriptProviderBase {
+    private static final long serialVersionUID = 523398435792363456L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/SoftCachingModuleScriptProvider.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/StrongCachingModuleScriptProvider.java [hunk 15]
```diff
@@ -17,12 +17,10 @@
...
  * @version $Id: StrongCachingModuleScriptProvider.java,v 1.3 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public class StrongCachingModuleScriptProvider extends CachingModuleScriptProviderBase
-{
-    private static final long serialVersionUID = 1L;
+public class StrongCachingModuleScriptProvider extends CachingModuleScriptProviderBase {
 
-    private final Map<String, CachedModuleScript> modules =
-        new ConcurrentHashMap<String, CachedModuleScript>(16, .75f, getConcurrencyLevel());
+    private static final long serialVersionUID = -8435427841788350515L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/StrongCachingModuleScriptProvider.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/UrlModuleSourceProvider.java [hunk 16]
```diff
@@ -31,14 +31,11 @@
  * @author Attila Szegedi
  * @version $Id: UrlModuleSourceProvider.java,v 1.4 2011/04/07 20:26:12 hannes%helma.at Exp $
  */
-public class UrlModuleSourceProvider extends ModuleSourceProviderBase
-{
-    private static final long serialVersionUID = 1L;
-
+public class UrlModuleSourceProvider extends ModuleSourceProviderBase {
+    private static final long serialVersionUID = -1319310258441679202L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/UrlModuleSourceProvider.java]: Compatibility.

File: src/org/mozilla/javascript/commonjs/module/provider/UrlModuleSourceProvider.java [hunk 17]
```diff
@@ -241,22 +238,18 @@ protected boolean entityNeedsRevalidation(Object validator) {
     }
 
     private static class URLValidator implements Serializable {
-        private static final long serialVersionUID = 1L;
-
-        private final URI uri;
-        private final long lastModified;
-        private final String entityTags;
-        private long expiry;
+    private static final long serialVersionUID = -832296617758674059L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/commonjs/module/provider/UrlModuleSourceProvider.java]: Compatibility.

File: xmlimplsrc/org/mozilla/javascript/xmlimpl/XMLLibImpl.java [hunk 18]
```diff
@@ -21,15 +21,16 @@
 import org.xml.sax.SAXException;
 
 public final class XMLLibImpl extends XMLLib implements Serializable {
-    private static final long serialVersionUID = 1L;
 
     //
     //    EXPERIMENTAL Java interface
     //
+   
+    private static final long serialVersionUID = 3519443689172208557L;
```
Comment (gbrail) [inline xmlimplsrc/org/mozilla/javascript/xmlimpl/XMLLibImpl.java]: Compatibility.

File: xmlimplsrc/org/mozilla/javascript/xmlimpl/XmlNode.java [hunk 19]
```diff
@@ -95,8 +97,6 @@ private static XmlNode copy(XmlNode other) {
         return createImpl( other.dom.cloneNode(true) );
     }
 
-    private static final long serialVersionUID = 1L;
```
Comment (gbrail) [inline xmlimplsrc/org/mozilla/javascript/xmlimpl/XmlNode.java]: Compatibility.

File: src/org/mozilla/javascript/NativeStringIterator.java (hunk 2 above)
Comment (oswetto) [reply src/org/mozilla/javascript/NativeStringIterator.java]: Yes, sir

File: src/org/mozilla/javascript/LazilyLoadedCtor.java [hunk 20]
```diff
@@ -15,13 +16,12 @@
  *
  * <p> This improves startup time and average memory usage.
  */
-public final class LazilyLoadedCtor implements java.io.Serializable {
-    private static final long serialVersionUID = 1L;
-
+public final class LazilyLoadedCtor implements Serializable {
+    private static final long serialVersionUID = 1426064956930791935L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/LazilyLoadedCtor.java]: This one too!

File: src/org/mozilla/javascript/NativeArrayIterator.java [hunk 21]
```diff
@@ -13,7 +13,7 @@
             VALUES
     }
 
-    private static final long serialVersionUID = 1L;
+    private static final long serialVersionUID = -2051497571989274095L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/NativeArrayIterator.java]: And this one!

File: src/org/mozilla/javascript/NativeWith.java [hunk 22]
```diff
@@ -14,13 +14,10 @@
  * It simply delegates every action to its prototype except
  * for operations on its parent.
  */
-public class NativeWith
-    implements Scriptable, SymbolScriptable, IdFunctionCall, Serializable {
+public class NativeWith implements Scriptable, SymbolScriptable, IdFunctionCall, Serializable {
+    private static final long serialVersionUID = 5581345585553901804L;
 
-    private static final long serialVersionUID = 1L;
```
Comment (gbrail) [inline src/org/mozilla/javascript/NativeWith.java]: And technically, we shouldn't remove this one either.

**Comment by gbrail** [comment]
Now that looks right. Thanks!

**Pull request merged**